apps/ledger-live-mobile/src/screens/WalletCentricSections/Allocations.tsx

```diff
@@ -51,13 +55,25 @@ const Allocations = () => {
   }, [navigation]);
 
   const distributionListFormatted = useMemo(() => {
-    if (distribution.list.length <= NUMBER_MAX_ALLOCATION_ASSETS_TO_DISPLAY) {
+    const displayedCurrencies = distribution.list.filter(asset => {
+      return asset.currency.type === "TokenCurrency"
+        ? !blacklistedTokenIds.includes(asset.currency.id)
+        : true;
+    });
```

Comment:
I found it weird to have a ternary operator here
I think it can be rewrite as something like, 
`asset.currency.type !== "TokenCurrency" || !blacklistedTokenIds.includes(asset.currency.id)`
WDYT (same for the other one)